app/models/thread.php

```diff
@@ -55,16 +41,82 @@ public function create(Comment $comment)
         }
 
         $db = DB::conn();
-        $db->begin();
 
-        $db->query('INSERT INTO thread SET title = ?, created = NOW()', 
-        array($this->title));
+        $params = array(
+            'user_id'  => $this->user_id,
+            'title'    => $this->title,
+            'category' => $this->category
+        );
+
+        $db->insert('thread', $params);
 
         $this->id = $db->lastInsertId();
 
         //write the first comment
         $comment->write($this->id);
 
-        $db->commit();
+    }
+
+    public function editThread()
+    {
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid Thread');
+        }
+
+        $db = DB::conn();
+        
+        $params = array(
+            'title'    => $this->title,
+            'category' => $this->category
+        );
+        $db->update('thread', $params, array('id' => $this->id));
+    }
+
+    public static function isOwner($user_id, $thread_id)
+    {
+        $db = DB::conn();
+
+        $row = $db->row('SELECT * FROM thread WHERE user_id = ? AND id = ?',
+        array($user_id, $thread_id));
+
+        return (bool) $row;
+    }
+
+    public function deleteThread()
```

Comment:
This is a bit redundant since you're already on the thread model.
Please change this to ```delete()```